openapi/spec/spec.py

```diff
@@ -67,20 +67,50 @@ def parse(self):
             self.parsed_schemas[schema.__name__] = parsed_schema
         return self.parsed_schemas
 
+    def field2json(self, field):
+        field = fields.as_field(field)
+        mapping = self._fields_mapping.get(field.type, None)
+        enum = None
+        if not mapping:
+            if is_subclass(field.type, Enum):
+                mapping = dict(type='string')
+                enum = [e.name for e in field.type]
+            elif is_subclass(field.type, List):
+                return self._list2json(field.type)
+            elif is_dataclass(field.type):
+                return self._get_schema_ref(field.type)
+            else:
+                raise InvalidTypeException(field.type)
+
+        json_property = {'type': mapping['type']}
+        meta = field.metadata
+        if meta.get(fields.DESCRIPTION):
+            json_property['description'] = meta.get(fields.DESCRIPTION)
+        format = meta.get(fields.FORMAT) or mapping.get(fields.FORMAT, None)
+        if format:
+            json_property[fields.FORMAT] = format
+        if enum:
+            json_property['enum'] = enum
+        validator = meta.get(fields.VALIDATOR)
+        if isinstance(validator, fields.Validator):
+            validator.openapi(json_property)
```

Comment:
It took me a while to figure out what `openapi` does here. Maybe rename or write a short docstring?